### PR TITLE
Implemented AppendCampaign Model

### PR DESCRIPTION
**Pull request by srs-vimalk**
Implemented AppendCampaign Request and Response model

### CODE REVIEW

**Review by srs-adamr — CHANGES_REQUESTED** [review]
Please correct the folder structure and placement of files.

File: LiveVox.NET/Models/Campaign/AppendCampaign/AppendCampaignResponse.cs
Comment (srs-adamr) [inline LiveVox.NET/Models/Campaign/AppendCampaign/AppendCampaignResponse.cs:1]: There is no response object, so please remove this.

**Review by srs-adamr — CHANGES_REQUESTED** [review]

File: LiveVox.NET/Models/Campaign/AppendCampaign/AppendCampaignResponse.cs
Comment (srs-vimalk) [reply LiveVox.NET/Models/Campaign/AppendCampaign/AppendCampaignResponse.cs:1]: I have removed this

Comment (srs-vimalk) [reply LiveVox.NET/Models/Campaign/AppendCampaign/AppendCampaignResponse.cs:1]: also changed the folder structure let me know if we need any changes

File: LiveVox.NET/Models/Campaign/Requests/AppendCampaignRequest.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
+﻿using System.ComponentModel.DataAnnotations;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using LiveVox.NET.Models.Base;
+using LiveVox.NET.Models.Campaign.Common;
+using RestSharp;
+
+namespace LiveVox.NET.Models.Campaign.AppendCampaign
```
Comment (srs-adamr) [inline LiveVox.NET/Models/Campaign/Requests/AppendCampaignRequest.cs]: Correct namespace

File: LiveVox.NET/Models/Campaign/Requests/CreateCampaignRequest.cs [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+using System.Text;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using System.Threading.Tasks;
+using LiveVox.NET.Models.Base;
+using LiveVox.NET.Models.Campaign.Common;
+using LiveVox.NET.Models.Campaign.Enumerations;
+using RestSharp;
+
+namespace LiveVox.NET.Models.Campaign.CreateCampaign
```
Comment (srs-adamr) [inline LiveVox.NET/Models/Campaign/Requests/CreateCampaignRequest.cs]: Correct namespace

File: LiveVox.NET/Models/Campaign/Responses/CreateCampaignResponse.cs [hunk 3]
```diff
@@ -0,0 +1,19 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Text.Json.Serialization;
+using System.Threading.Tasks;
+using LiveVox.NET.Models.Base;
+
+namespace LiveVox.NET.Models.Campaign.CreateCampaign
```
Comment (srs-adamr) [inline LiveVox.NET/Models/Campaign/Responses/CreateCampaignResponse.cs]: Correct namespace

**Comment by srs-vimalk** [comment]
Implemented Response and Request model

1. AppendCampaignRecord
2. CreateCampaign
3. DeactivateCampaign
4. FindMatchingCampaigns
5. FindMatchingFinishedCalls
6. GetCampaignListInfo
7. ListCampaigns
8. ListCampaignTransactions
9. ReadCampaign
10. UpdateCampaign
11. UpdateCampaignState
﻿committed changes

File: LiveVox.NET/Models/Campaign/Common/AccountTransactionEntry.cs
Comment (srs-adamr) [inline LiveVox.NET/Models/Campaign/Common/AccountTransactionEntry.cs:1]: This file is missing it's Json attributes.

File: LiveVox.NET/Models/Campaign/Common/CampaignFIle.cs
Comment (srs-adamr) [inline LiveVox.NET/Models/Campaign/Common/CampaignFIle.cs:1]: This file is missing it's json attributes

File: LiveVox.NET/Models/Campaign/Common/Consumer.cs
Comment (srs-adamr) [inline LiveVox.NET/Models/Campaign/Common/Consumer.cs:1]: This file is missing it's json attributes

File: LiveVox.NET/Models/Campaign/Common/Consumer.cs [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+        /// <summary>
+        /// Gets or sets the last name of the consumer.
+        /// </summary>
+        public string LastName { get; set; }
+
+        /// <summary>
+        /// Gets or sets the date of birth of the consumer.
+        /// </summary>
+        public DateTime? DateOfBirth { get; set; }
```
Comment (srs-adamr) [inline LiveVox.NET/Models/Campaign/Common/Consumer.cs]: Double check this format, most likely it is also a Unix MS timestamp

File: LiveVox.NET/Models/Campaign/Requests/ListCampaignTransactionsRequest.cs [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+using System.Text.Json;
+using System.Threading.Tasks;
+
+namespace LiveVox.NET.Models.Campaign.Requests
+{
+    public class ListCampaignTransactionsRequest : ILiveVoxRequest
+    {
+        public string? Category { get; set; } = "campaign";
+        public string? Resource { get; set; } = "campaigns/{id}/transactionList";
+        public string? TransactionList { get; set; } = "transactionList";
```
Comment (srs-adamr) [inline LiveVox.NET/Models/Campaign/Requests/ListCampaignTransactionsRequest.cs]: What is this for?

File: LiveVox.NET/Models/Campaign/Responses/CreateCampaignResponse.cs
Comment (srs-adamr) [inline LiveVox.NET/Models/Campaign/Responses/CreateCampaignResponse.cs:1]: Json attributes missing

File: LiveVox.NET/Models/Campaign/Responses/ReadCampaignResponse.cs
Comment (srs-adamr) [inline LiveVox.NET/Models/Campaign/Responses/ReadCampaignResponse.cs:1]: Json Attrbutes missing

**Review by srs-adamr — CHANGES_REQUESTED** [review]
Please review changes, ensure all requests/responses properties that are being used in a body have their json attributes.

File: LiveVox.NET/Models/Campaign/Responses/ReadCampaignResponse.cs
Comment (srs-vimalk) [reply LiveVox.NET/Models/Campaign/Responses/ReadCampaignResponse.cs:1]: I had made the changes, can you please review it

File: LiveVox.NET/Models/Campaign/Requests/AppendCampaignRequest.cs [hunk 6]
```diff
@@ -9,7 +9,9 @@ namespace LiveVox.NET.Models.Campaign.Requests
 {
     public class AppendCampaignRequest : ILiveVoxRequest
     {
+        [JsonIgnore]
         public string? Category { get; set; } = "campaign";
+        [JsonIgnore]
         public string? Resource { get; set; } = "campaigns/{id}/transactions";
         public Method RequestType { get; set; } = Method.Put;
```
Comment (srs-adamr) [inline LiveVox.NET/Models/Campaign/Requests/AppendCampaignRequest.cs:17]: RequestType also needs ignored.

**Review by srs-adamr — CHANGES_REQUESTED** [review]
Still need some namespace corrections, and JsonIgnore for RequestTypes

**Review by srs-adamr — APPROVED** [review]
Looks good.

**Review by srs-binnys — APPROVED** [review]

**Pull request merged**